### ISSUE TITLE
[PB-35980] Folder Filter Returning Duplicate Resources

### ISSUE THREAD

**Issue by Paulo1402** [passbolt/passbolt_api#523]
## Folder Filter Returning Duplicate Resources
Passbolt Version: 4.9.1
Platform and Target: 
-- Operating system: Ubuntu 20.04.6 LTS
-- PHP:  7.4.3-4ubuntu2.24
-- Web server: Nginx 1.18.0
-- Database server: MySQL 15.1

### What you did
When I fetch `/resources.json` with the `has-parent` filter, it returns a lot of duplicate entries.

**Request example:**
_GET https://base_url/resources.json?filter%5Bhas-parent%5D=3dda7223-35a4-4010-b86f-d2a0530e3ddf_

### What happened
The filter correctly returns only the resources within the folder, but each resource is returned multiple times. It seems the results are multiplied by the number of users who can access that resource. For example, if I have 40 resources in a folder and share it with a group of 8 members, using the `has-parent` filter returns 320 resources, with each resource duplicated 8 times.

![image](https://github.com/user-attachments/assets/7d683f31-7a8d-45be-a9f1-ba9248a87ce7)

![image](https://github.com/user-attachments/assets/819dbeeb-86d7-47c9-858b-9e8192cd63bb)

![image](https://github.com/user-attachments/assets/4464242a-97fc-47dd-8d0d-53496b1db9e4)

I noticed that if a folder is shared with more than one group and there is a user who is in both groups, the entry will only return once for that user.

### What you expected to happen
I expected the result to include only the resources within the folder, with each entry being returned once.

**Comment by stripthis** [comment]
Thank @Paulo1402 we'll have a look. For the avoidance of doubt can make sure you make sure the data is clean:
```
sudo su -s /bin/bash -c './bin/cake passbolt cleanup' www-data
sudo su -s /bin/bash -c './bin/cake passbolt datacheck' www-data
```

**Comment by Paulo1402** [comment]
> Thank @Paulo1402 we'll have a look (PB-35980) For the avoidance of doubt can make sure you make sure the data is clean:
> 
> ```
> sudo su -s /bin/bash -c './bin/cake passbolt cleanup' www-data
> sudo su -s /bin/bash -c './bin/cake passbolt datacheck' www-data
> ```

Of course! I just ran these commands and didn't see anything wrong, all the data seems to be fine.

**Comment by ishanvyas22** [comment]
Hey @Paulo1402, we are able to reproduce the issue and working on the fix. We will update you once it gets released (ideally with v4.10).

**Comment by Paulo1402** [comment]
> Hey @Paulo1402, we are able to reproduce the issue and working on the fix. We will update you once it gets released (ideally with v4.10).

Thank you @ishanvyas22! I will be waiting for you guys!
